scripts/lib/CIME/SystemTests/system_tests_compare_two.py

```diff
@@ -161,28 +193,38 @@ def run_phase(self, success_change=False):  # pylint: disable=arguments-differ
         Runs both phases of the two-phase test and compares their results
         If success_change is True, success requires some files to be different
         """
+        first_phase = self._case1.get_value("RESUBMIT") == 1
+        run_type = self._case1.get_value("RUN_TYPE")
 
         # First run
-        logger.info('Doing first run: ' + self._run_one_description)
-        self._activate_case1()
-        run_type = self._case1.get_value("RUN_TYPE")
-        self.run_indv(suffix = self._run_one_suffix)
+        if not self._multisubmit or first_phase:
+            logger.info('Doing first run: ' + self._run_one_description)
+            self._activate_case1()
+            self._case_one_custom_prerun_action()
+            self.run_indv(suffix = self._run_one_suffix)
+            self._case_one_custom_postrun_action()
 
         # Second run
-        logger.info('Doing second run: ' + self._run_two_description)
-        self._activate_case2()
-        # we need to make sure run2 is properly staged.
-        if run_type != "startup":
-            check_case(self._case2, self._caseroot2)
-        self._force_case2_settings()
-        self.run_indv(suffix = self._run_two_suffix)
-
-        # Compare results
-        # Case1 is the "main" case, and we need to do the comparisons from there
-        self._activate_case1()
-        self._link_to_case2_output()
+        if not self._multisubmit or not first_phase:
+            with self._case2:
```

Comment:
@billsacks, I understand your confusion; this one took me a while to figure out. Your a and b questions seem like the same question so I'll answer them together.

(a/b): This was exposed by the case.flush() in the NODEFAIL test. This flush will fail if the case is read-only.
(c): case1 is already in a writeable state. Recall that case1 is the original case while case2 is the close. That means case1 has already been opened with a with statement (if you look at the command-line tools in scripts, scripts/Tools, that's generally the first thing they do (take a caseroot and open a Case object with a with statement).